x/cqlbuilder/execer.go

```diff
@@ -0,0 +1,71 @@
+package cqlbuilder
+
+import (
+	"context"
+
+	"github.com/upfluence/cql"
+)
+
+type CASScanner interface {
+	ScanCAS(map[string]interface{}) (bool, error)
+}
+
+type errCASScanner struct{ error }
+
+func (ecs errCASScanner) ScanCAS(map[string]interface{}) (bool, error) {
+	return false, ecs.error
+}
+
+type casScanner struct {
+	sc cql.CASScanner
+	ks []string
+}
+
+func (cs *casScanner) ScanCAS(qvs map[string]interface{}) (bool, error) {
+	vs := make([]interface{}, len(cs.ks))
+
+	for i, k := range cs.ks {
+		v, ok := qvs[k]
+
+		if !ok {
+			return false, ErrMissingKey{Key: k}
+		}
+
+		vs[i] = v
+	}
+
+	return cs.sc.ScanCAS(vs...)
+}
+
+type Execer interface {
+	Exec(context.Context, map[string]interface{}) error
+	ExecCAS(context.Context, map[string]interface{}) CASScanner
+}
+
+type execer struct {
```

Comment:
Cassandra noob here,

What's the difference between `Exec` and `ExecCAS` ? Expecting a scanner to deal with some output ? If so what does CAS stands for ?